### PR TITLE
Relax german-sysadmin integer literal check

### PR DESCRIPTION
**Pull request by angelikatyborska**
Resolves https://github.com/exercism/elixir-analyzer/issues/384

The check is relaxed - now just one usage of `?ß`, `?ä`, `?ö`, `?ü`, `?_`, `?a`, or `?z` in the solution is enough, as long as the corresponding integer literals aren't found.

A trade-off: this will produce false-positives in scenarios in which people randomly include one of those numbers in the code, including in comments, e.g. `?ß # 223`

### CODE REVIEW

**Comment by dabaer** [comment]
RE the false positives, you could reduce the surface area by removing comments from the checked code:

```elixir
check(%Source{code_string: code_string}) do
  question_mark_code_points = ["?ß", "?ä", "?ö", "?ü", "?_", "?a", "?z"]
  integer_literal_code_points = ["223", "228", "246", "252", "95", "97", "122"]

  code_string = code_string |> Code.string_to_quoted!() |> Macro.to_string()

  Enum.any?(question_mark_code_points, &String.contains?(code_string, &1)) &&
    not Enum.any?(integer_literal_code_points, &String.contains?(code_string, &1))
end
```

File: mix.lock [hunk 1]
```diff
@@ -12,6 +12,6 @@
   "metrics": {:hex, :metrics, "1.0.1", "25f094dea2cda98213cecc3aeff09e940299d950904393b2a29d191c346a8486", [:rebar3], [], "hexpm", "69b09adddc4f74a40716ae54d140f93beb0fb8978d8636eaded0c31b6f099f16"},
   "mimerl": {:hex, :mimerl, "1.2.0", "67e2d3f571088d5cfd3e550c383094b47159f3eee8ffa08e64106cdf5e981be3", [:rebar3], [], "hexpm", "f278585650aa581986264638ebf698f8bb19df297f66ad91b18910dfc6e19323"},
   "parse_trans": {:hex, :parse_trans, "3.3.1", "16328ab840cc09919bd10dab29e431da3af9e9e7e7e6f0089dd5a2d2820011d8", [:rebar3], [], "hexpm", "07cd9577885f56362d414e8c4c4e6bdf10d43a8767abb92d24cbe8b24c54888b"},
-  "ssl_verify_fun": {:hex, :ssl_verify_fun, "1.1.7", "354c321cf377240c7b8716899e182ce4890c5938111a1296add3ec74cf1715df", [], [], "hexpm", "fe4c190e8f37401d30167c8c405eda19469f34577987c76dde613e838bbc67f8"},
+  "ssl_verify_fun": {:hex, :ssl_verify_fun, "1.1.7", "354c321cf377240c7b8716899e182ce4890c5938111a1296add3ec74cf1715df", [:make, :mix, :rebar3], [], "hexpm", "fe4c190e8f37401d30167c8c405eda19469f34577987c76dde613e838bbc67f8"},
```
Comment (dabaer) [inline mix.lock:15]: I noticed prior to the 1.15 update `[:make, :mix, :rebar3]` was in this line, but it got removed. Weird that it is now back. I can't seem to find much information on the format of the `mix.lock` file to understand this behaviour.

**Comment by angelikatyborska** [comment]
@dabaer The reason we're doing this check on the string in the first place is the same reason your suggestion wouldn't work: the `?` syntax is just syntactic sugar that gets completely stripped when producing an AST:

```elixir
iex(6)> "if x == ?A, do: ?B, else: 0"
"if x == ?A, do: ?B, else: 0"

iex(7)> v |> Code.string_to_quoted!()
{:if, [line: 1],
 [{:==, [line: 1], [{:x, [line: 1], nil}, 65]}, [do: 66, else: 0]]}

iex(8)> v |> Macro.to_string()
"if(x == 65) do\n  66\nelse\n  0\nend"
```

**Review by jiegillet — APPROVED** [review]
I'm good with this.

Maybe one way to reduce the false positives a little bit would be to forbid the use of any of the `integer_literal_code_points` except if their question mark equivalent is also used.

```elixir
# OK
?ß -> ~c"ss"
# Not OK
223 -> ~c"ss"
# Still OK because both are used
 ?ß -> ~c"ss" # 223
```

**Comment by angelikatyborska** [comment]
@jie like this [`7839ac0` (#389)](https://github.com/exercism/elixir-analyzer/pull/389/commits/7839ac0bf2dfe1d50a35e3457f9fec1a06538039) ?

**Comment by jiegillet** [comment]
@angelikatyborska yes, looks good! (you called the wrong jie in the process though haha)

**Comment by angelikatyborska** [comment]
Hahaha, oh no 🙈 that explains why you didn't react

**Pull request merged**